nucliadb_vectors/src/data_point/ops_hnsw.rs

```diff
@@ -157,7 +157,9 @@ impl<'a, DR: DataRetriever> HnswOps<'a, DR> {
                 // Our best candidate is worse than the min score
                 (Some(Cnx(_, cs)), _) if cs < self.tracker.min_score() => break,
                 // Our best candidate is worse than the worst result
-                (Some(Cnx(_, cs)), Some(Reverse(Cnx(_, ws)))) if cs < ws => break,
+                (Some(Cnx(_, cs)), Some(Reverse(Cnx(_, ws)))) if cs < ws && results.len() == k_neighbours => {
```

Comment:
Since this is filtered BFS search, we can only stop here if we have all that we need.

Reply:
Sorry  I was wrong, this is actually ok 🙏